backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -276,6 +278,35 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
         }
     }
 
+    private fun <R> switchTo(irFunction: IrFunction, block: () -> R): R? {
+        val oldCodeContext = currentCodeContext
+
+        try {
+            if ((currentCodeContext as? FunctionScope)?.declaration == irFunction
+                    || (currentCodeContext as? ReturnableBlockScope)?.returnableBlock?.inlineFunctionSymbol?.owner == irFunction
+                    || (currentCodeContext as? FunctionScope)?.declaration == irFunction)
+                return block()
+            currentCodeContext = (currentCodeContext as? InnerScope)?.outerContext ?: return null
```

Comment:
So there is a situation when we don't call `block`? Is it correct? May be better would be to throw an error?

Reply:
correctness of code generation  doesn't degrade, because we process this case in `generateVariable`, but what relates to debug information generation, that yes it'll be inaccurate.

Reply:
Ah, i see.